lib/experimental-default-theme.json

```diff
@@ -129,5 +129,12 @@
 				"dropCap": false
 			}
 		}
+	},
+	"core/cover": {
+		"styles": {
+			"spacing": {
+				"padding": 16
```

Comment:
This is how we'd absorb the existing CSS properties of the blocks. A few thoughts on this:

First, I think I'd wait post-5.5 to do this kind of change, so we have a full cycle to test in the plugin, as this requires open-up theme.json to production. [Related conversation](https://github.com/WordPress/gutenberg/pull/22929): for the link color feature to work, we need to expose the presets declared by the theme via `add_theme_support` as CSS variables. We may be able to do this by only shipping to WordPress core the function that transforms the theme support declarations into CSS variables, without the need of shipping theme.json just yet. I thought I'd connect this two convos for perspective.

A second point to talk about is that, ideally, we can reference [this sort of variables](https://github.com/WordPress/gutenberg/blob/master/packages/base-styles/_variables.scss#L27) (`$grid-unit-20`) via theme.json. We can also leave as it is, hardcoded values, but I think it'll be easy if we can use "named variables". Two ways we can do this are:

1. We add the grid unit values as if they were any other presets (font-size, color) in theme.json, and theme authors will reference them as they do with any other preset: 

```
{
  "global": {
    "presets": {
      "padding": [
          {
              "slug": "unit",
              "value": 8
          },
          {
              "slug": "unit-20",
              "value": 16
          }
      ]
    }
  },
  "core/cover": {
    "styles": {
      "spacing": {
        "padding": "var(--wp--preset--padding--unit-20)"
      }
    }
  }
}
```

2. We create a mechanism to resolve the variable at render time. We need to mark the values to be resolved as well as having those variables available somewhere. Let's pretend that all values that are enclosed by `<>` should be resolved to the variable and that we expose them via a `variables` subkey of the global scope (I'm not attached to this specific format). Theme authors could then do:

```
{
  "global": {
    "variables": {
      "grid-unit": 8,
      "grid-unit-20": 16
    }
  },
  "core/cover": {
    "styles": {
      "spacing": {
        "padding": "<global.variables.grid-unit-20>"
      }
    }
  }
}
```

Reply:
> We create a mechanism to resolve the variable at render time.

Is the intent to ensure compat for browsers with no `var()` support? Or is it something else?

Reply:
That's something that crossed my mind, yes, although we're already using CSS vars in other places (presets), so we need to provide fallbacks for this sort of situation anyway.

One advantage the second approach has it that it doesn't pollute the global space with all the variables that are present in the [base-styles](https://github.com/WordPress/gutenberg/tree/master/packages/base-styles). However, I don't know how much a problem that would be in practice. On the other hand, the second approach requires a specific syntax, which I consider a disadvantage. I like how the first approach requires fewer concepts to learn: leaf values just use regular CSS syntax.

Reply:
Yes, it's not an easy decision. 

> it doesn't pollute the global space with all the variables that are present in the [base-styles](https://github.com/WordPress/gutenberg/tree/master/packages/base-styles).

Can this be mitigated by properly namespacing them, even if they are globally available?

Reply:
More brain-dumping: we'll need to expose the theme.json via some REST endpoint for mobile to be able to use/expose this data in the apps. One consideration we should make is how to balance theme authors' needs (use already-known CSS syntax at the leaf nodes) vs mobile code's needs (use parseable syntax at the leaf nodes).

Reply:
Thanks @nosolosw for your consideration on mobile end. It should be possible to evaluate formulas with basic operators in it but not sure how to evaluate the css functions on mobile side, I don't think there's an easy way :( cc @dratwas @maxme @hypest if you want to bring a different point of view.

Reply:
> not sure how to evaluate the css functions on mobile side

Can't say I have good insight on the matter, but I wonder, what CSS functions are we perhaps talking about? Talking about the `var()` specifically or others too?

Reply:
@pinarol has been helping me levelling up a bit on styling for the mobile app. I think we can move the discussion over this issue https://github.com/WordPress/gutenberg/issues/24165 which I also listed in the tracking issue for the work related to the block style system https://github.com/WordPress/gutenberg/issues/20331

Reply:
I prefer the usage of CSS syntax "var(--wp--preset--padding--unit-20)". We would need to parse things anyway no matter if it is "var(--wp--preset--padding--unit-20)" or "<global.variables.grid-unit-20>". And in that case, I would prefer if we used what developers already know instead of needing to specify a new syntax and a parser for it.

So I guess things could look like this:
```
{
  "global": {
    "variables": {
      "grid-unit": 8,
      "grid-unit-20": 16
    }
  },
  "core/cover": {
    "styles": {
      "spacing": {
        "padding": "var(--wp--global--variables--grid-unit-20)"
      }
    }
  }
}
```

Reply:
I wouldn't worry if it was just about `var()` actually, as you said we need to parse things anyway. But the possibility of using any kind of css function(like `calc()`) is the main blocker for native mobile. Convo going on here: https://github.com/WordPress/gutenberg/issues/24165

We'll be starting to actively work on GSS for native mobile in a couple of weeks so hopefully we'll have more progress on that issue. ✌️